### ISSUE TITLE
[BUG] bundle.Download() should have a context.Context argument

### ISSUE THREAD

**Issue by cfergeau** [crc-org/crc#3998]
It's not possible to programmatically cancel bundle downloads. After using `start` with the REST API, if the bundle has to be downloaded, then it's not possible to use `stop` while the bundle is being downloaded.

**Comment by cfergeau** [comment]
Reproducer:
On a system where the bundle is not downloaded/unpacked:
```
$ curl --unix-socket /var/home/teuf/.crc/crc-http.sock  http://unix/api/start
# the start command hangs waiting for the bundle to be downloaded

# in another terminal
$ curl --unix-socket /var/home/teuf/.crc/crc-http.sock  http://unix/api/stop
cannot abort startup sequence quickly enough
```

It would be nice if, when we call `stop` while a bundle is being downloaded/uncompressed, it's still possible to call `stop`, and cancel the download/uncompression.
Hopefully we can use the `context.Context` passed to `pkg/crc/machine.Start` to cancel all of this.

**Comment by rohanKanojia** [comment]
@redbeam @cfergeau : Hello, is this issue fixed by #4405

**Comment by cfergeau** [comment]
Yes, thanks for spotting this.
